packages/docusaurus-plugin-content-docs/index.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const path = require('path');
+const globby = require('globby');
+const {getSubFolder, idx, normalizeUrl} = require('@docusaurus/utils');
+const createOrder = require('./src/order');
+const loadSidebars = require('./src/sidebars');
+const processMetadata = require('./src/metadata');
+
+const DEFAULT_OPTIONS = {
+  metadataKey: 'docsMetadata',
+  metadataFileName: 'docsMetadata.json',
+  path: 'docs', // Path to data on filesystem, relative to site dir.
+  routeBasePath: 'docs', // URL Route.
+  include: ['*.md', '*.mdx'], // Extensions to include.
+  // TODO: Read from props rather than hardcoded sidebar.json.
+  sidebar: [], // Sidebar configuration for showing a list of documentation pages.
+  docLayoutComponent: '@theme/Doc',
+  docItemComponent: '@theme/DocBody',
+};
+
+class DocusaurusPluginContentDocs {
+  constructor(opts, context) {
+    this.options = {...DEFAULT_OPTIONS, ...opts};
+    this.context = context;
+    this.contentPath = path.resolve(this.context.siteDir, this.options.path);
+  }
+
+  getName() {
+    return 'docusaurus-plugin-content-docs';
+  }
+
+  getPathsToWatch() {
+    return [this.contentPath];
+  }
+
+  // Fetches blog contents and returns metadata for the contents.
+  async loadContent() {
+    const {include, routeBasePath} = this.options;
+    const {siteDir, env, siteConfig, cliOptions = {}} = this.context;
+    const {skipNextRelease} = cliOptions;
+    const docsDir = this.contentPath;
+
+    // @tested - load all sidebars including versioned sidebars
+    const docsSidebars = loadSidebars({siteDir, env});
+
+    // @tested - build the docs ordering such as next, previous, category and sidebar
+    const order = createOrder(docsSidebars);
+
+    // Settle versions & translations from environment.
+    const translationEnabled = idx(env, ['translation', 'enabled']);
+    const enabledLanguages =
+      translationEnabled && idx(env, ['translation', 'enabledLanguages']);
+    const enabledLangTags =
+      (enabledLanguages && enabledLanguages.map(lang => lang.tag)) || [];
+    const defaultLangTag = idx(env, ['translation', 'defaultLanguage', 'tag']);
+    const versioningEnabled = idx(env, ['versioning', 'enabled']);
+    const versions =
+      (versioningEnabled && idx(env, ['versioning', 'versions'])) || [];
+
+    // Prepare metadata container.
+    const docs = {};
+
+    if (!(versioningEnabled && skipNextRelease)) {
+      // Metadata for default docs files.
+      const docsFiles = await globby(include, {
```

Comment:
The reason why the test fails is because it wasn't able to load the docs in a subdirectory.
The build on v2 succeed because we don't have a subdirectory docs. (we have flat doc structure)

Previously:
https://github.com/facebook/Docusaurus/blob/c33e874e1c0abbe0d9c7cab8ce93882746e5d293/packages/docusaurus/lib/load/docs/index.js#L44

Now it's like
```js
const docsFiles = await globby(['*.md'], {
      cwd: docsDir,
    });
```

Reply:
Great catch! Thanks, this is the main reason.